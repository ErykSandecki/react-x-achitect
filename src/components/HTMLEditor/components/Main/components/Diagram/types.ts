// types
import { BlockType, PortType, T2DCoordinates, TBlock, TConnector } from 'types';

export type TDraggableConnectorSource = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnectorTarget = {
  blockId: string;
  blockType: BlockType;
  portType: PortType;
};

export type TDraggableConnector = {
  showIndicator: boolean;
  source?: TDraggableConnectorSource;
  sourceCoordinates: T2DCoordinates;
  target?: TDraggableConnectorTarget;
  targetCoordinates: T2DCoordinates;
};

export type TPossiblePort = T2DCoordinates & {
  blockId: string;
  blockType: BlockType;
  height: number;
  portType: PortType;
  width: number;
};

export type TShadowConnector = Pick<
  TConnector,
  'joinOperator' | 'relatedWithBlockId' | 'towardsTo'
> & {
  sourceBlock: Partial<TBlock> & { portType: PortType };
  targetBlock: Partial<TBlock> & { portType: PortType };
};
