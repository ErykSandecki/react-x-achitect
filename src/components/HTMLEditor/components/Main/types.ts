import { MutableRefObject } from 'react';

export type TMainRefs = {
  blocksPanelRef: MutableRefObject<HTMLDivElement>;
  diagramRef: MutableRefObject<HTMLDivElement>;
  panelPropertiesRef: MutableRefObject<HTMLDivElement>;
};
