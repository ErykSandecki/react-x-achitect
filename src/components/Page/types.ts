import { ReactNode } from 'react';

// types
import { PageType } from './enums';

export type TPageType = Record<PageType, string>;

export type TRestProps<T> = {
  children: ReactNode;
  pageType?: T;
  translationNameSpace?: string;
  withBackground?: boolean;
};
