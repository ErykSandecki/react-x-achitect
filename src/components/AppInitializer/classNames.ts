export const className = 'AppInitializer';

export const classNames = {
  [className]: className,
  logoWrapper: `${className}__logo-wrapper`,
  squaresWrapper: `${className}__squares-wrapper`,
  firstSquareWrapper: `${className}__first-square-wrapper`,
  firstSquare: `${className}__first-square`,
  secondSquareWrapper: `${className}__second-square-wrapper`,
  secondSquare: `${className}__second-square`,
  logo: `${className}__logo`,
  companyName: `${className}__company-name`,
};
