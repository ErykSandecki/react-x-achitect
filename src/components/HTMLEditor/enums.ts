export enum BlockHeight {
  small = 50,
  medium = 70,
  large = 100,
}

export enum BlockWidth {
  small = 50,
  medium = 200,
  large = 350,
}

export enum MouseMode {
  comment = 'comment',
  default = 'default',
  menu = 'menu',
  move = 'move',
}
