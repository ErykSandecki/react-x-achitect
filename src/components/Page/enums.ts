export enum PageType {
  blank = 'blank',
  edit = 'edit',
  layout = 'layout',
}
