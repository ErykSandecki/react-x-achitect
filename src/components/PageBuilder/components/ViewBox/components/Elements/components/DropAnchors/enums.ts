export enum DropAnchorsPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
