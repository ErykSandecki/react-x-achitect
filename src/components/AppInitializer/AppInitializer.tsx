import { FC, useContext, useEffect } from 'react';

// core
import { Context } from 'core/ContextProvider/ContextProvider';

// hooks
import { useDispatchMany } from 'hooks/useDispatchMany/useDispatchMany';

// others
import { APPLICATION_NAME } from 'constant/constants';
import { className, classNames } from './classNames';

// store
import { appInit, initLanguage } from 'store/appInitializer/actions';

// styles
import styles from './styles/app-initializer.scss';

// types
import { Theme } from 'types/enums';

const AppInitializer: FC = () => {
  const { theme } = useContext(Context);
  const dispatchMany = useDispatchMany();
  const isDarkMode = theme === Theme.dark;

  useEffect(() => {
    dispatchMany(appInit(), initLanguage());
  }, []);

  return <div className={styles[classNames[className]]}></div>;
};

export default AppInitializer;
