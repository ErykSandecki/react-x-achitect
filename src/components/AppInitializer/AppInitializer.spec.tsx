import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import AppInitializer from './AppInitializer';

// core
import { ContextProvider } from 'core/ContextProvider/ContextProvider';

// mocks
import { appInitializerStateMock } from 'test/mocks/reducer/appInitializerMock';

// store
import { configureStore } from 'store/store';

// types
import { Theme } from 'types/enums';

const stateMock = {
  ...appInitializerStateMock,
};

describe('AppInitializer snapshots', () => {
  it('should render AppInitializer', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <AppInitializer />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render in dark mode', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ContextProvider theme={Theme.dark}>
          <AppInitializer />
        </ContextProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
