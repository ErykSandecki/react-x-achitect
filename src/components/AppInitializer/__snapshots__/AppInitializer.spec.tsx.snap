// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInitializer snapshots should render AppInitializer 1`] = `
<DocumentFragment>
  <div
    class="AppInitializer"
  >
    <div
      class="AppInitializer__logo-wrapper"
    >
      <div
        class="AppInitializer__squares-wrapper"
      >
        <div
          class="AppInitializer__first-square-wrapper"
        >
          <div
            class="AppInitializer__first-square"
          />
        </div>
        <div
          class="AppInitializer__second-square-wrapper"
        >
          <div
            class="AppInitializer__second-square"
          />
        </div>
      </div>
      <div
        class="AppInitializer__logo"
      />
    </div>
    <h1
      class="AppInitializer__company-name typography_Typography__Hvofe typography_Typography--light__ma6Vx typography_Typography--semi-bold__o4iwT typography_Typography--normal__2w6Rv typography_Typography--h1__NYzWT typography_Typography--without-margin__uJld2"
      data-test-text=""
      style="text-align: inherit;"
    >
      CryptoTradingHub
    </h1>
  </div>
</DocumentFragment>
`;

exports[`AppInitializer snapshots should render in dark mode 1`] = `
<DocumentFragment>
  <div
    class="AppInitializer"
  >
    <div
      class="AppInitializer__logo-wrapper"
    >
      <div
        class="AppInitializer__squares-wrapper"
      >
        <div
          class="AppInitializer__first-square-wrapper"
        >
          <div
            class="AppInitializer__first-square"
          />
        </div>
        <div
          class="AppInitializer__second-square-wrapper"
        >
          <div
            class="AppInitializer__second-square"
          />
        </div>
      </div>
      <div
        class="AppInitializer__logo"
      />
    </div>
    <h1
      class="AppInitializer__company-name typography_Typography__Hvofe typography_Typography--dark__3M8RF typography_Typography--semi-bold__o4iwT typography_Typography--normal__2w6Rv typography_Typography--h1__NYzWT typography_Typography--without-margin__uJld2"
      data-test-text=""
      style="text-align: inherit;"
    >
      CryptoTradingHub
    </h1>
  </div>
</DocumentFragment>
`;
